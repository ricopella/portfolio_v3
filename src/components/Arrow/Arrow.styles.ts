import styled from '@emotion/styled'
import { AnchorLink } from 'gatsby-plugin-anchor-links'
import { BREAKPOINTS, COLORS } from '../../styles/variables'
import { keyframes } from '@emotion/core'

// Originally from: https://codepen.io/TommiTikall/pen/xZwpGR
const Arrow = styled.div`
  left: calc(50% - 1.5rem);
  opacity: 0;
  position: absolute;
  top: 85%;
  transform-origin: 50% 50%;
  transform: translate3d(-50%, -50%, 0);

  &:before,
  &:after {
    background: ${props => props.theme.actionColor};
    content: '';
    display: block;
    height: 0.1875rem;
    left: 0;
    position: absolute;
    top: 0;
    width: 1.875rem;
  }

  @media (max-width: ${BREAKPOINTS.smRem}) {
    top: 75%;
  }

  &:before {
    transform: rotate(45deg) translateX(-23%);
    transform-origin: top left;
  }

  &:after {
    transform: rotate(-45deg) translateX(23%);
    transform-origin: top right;
  }
`

const AnimateArrow = keyframes`
0% {
  opacity: 0;
  top: 80%;
}
70% {
  opacity: 1;
}
100% {
  opacity: 0;
}
`

const AnimateMobileArrow = keyframes`
0% {
  opacity: 0;
  top: 70%;
}
70% {
  opacity: 1;
}
100% {
  opacity: 0;
}
`

const ArrowFirst = styled(Arrow)`
  animation: ${AnimateArrow} 2s ease-in-out infinite;

  @media (max-width: ${BREAKPOINTS.smRem}) {
    animation: ${AnimateMobileArrow} 2s 1s ease-in-out infinite;
  }
`

const ArrowSecond = styled(Arrow)`
  animation: ${AnimateArrow} 2s 1s ease-in-out infinite;

  @media (max-width: ${BREAKPOINTS.smRem}) {
    animation: ${AnimateMobileArrow} 2s 1s ease-in-out infinite;
  }
`

const ArrowMask = styled(AnchorLink)`
  background-color: transparent;
  border: none;
  color: transparent;
  height: 5rem;
  left: 48%;
  position: absolute;
  top: 78%;
  width: 5rem;
  z-index: 5;
  opacity: 0;

  @media (max-width: ${BREAKPOINTS.smRem}) {
    left: 37%;
    position: absolute;
    top: 68%;
  }
`

export default {
  Arrow,
  ArrowFirst,
  ArrowMask,
  ArrowSecond,
}
