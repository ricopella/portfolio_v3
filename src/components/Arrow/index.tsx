import React from 'react'
import Styled from './Arrow.styles'

const Arrow = () => (
  <div>
    <Styled.ArrowFirst />
    <Styled.ArrowSecond />
    <Styled.ArrowMask to="/#about-me">Scroll Down</Styled.ArrowMask>
  </div>
)

export default Arrow
