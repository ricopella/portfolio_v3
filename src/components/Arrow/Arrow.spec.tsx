import renderer from 'react-test-renderer'
import React from 'react'
import Arrow from './index'

describe('<Arrow />', () => {
  it('should match Arrow snapshot', () => {
    const internalTree = renderer.create(<Arrow />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })
})
