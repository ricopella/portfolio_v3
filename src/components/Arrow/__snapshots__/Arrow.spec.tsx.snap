// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> should match Arrow snapshot 1`] = `
<div>
  <div
    className="css-1o9ii3w"
  />
  <div
    className="css-1rmi5y1"
  />
</div>
`;
